StringList/StringList.cpp

```diff
@@ -0,0 +1,357 @@
+#include "stringlist.h"
+#pragma warning(disable: 4996)
+
+#include <stdlib.h>
+#include <string.h>
+
+void StringListInit(char*** list)
+{
+    // creating dummy node
+    *list = static_cast<char**>(malloc(sizeof(char**) * 2));
+
+    if (*list == nullptr)
+        exit(-1);
+
+    char** cur = *list;
+    cur[0] = nullptr;
+    cur[1] = nullptr;
+}
+
+void StringListDestroy(char*** list)
+{
+    if (!list)
+        return;
+
+    char*** current = reinterpret_cast<char***>(*list);
+     ++current = reinterpret_cast<char***>(*current);
+    char*** del = list;
+
+    // deleting dummy node
+    free(del);
+
+    if (!current)
+        return;
+
+    del = current;
+
+    while (true)
+    {
+        // moving to next node
+        ++current = reinterpret_cast<char***>(*current);
+
+        // deleting string
+        free(del[0]);
+        // deleting node
+        free(del);
+
+        if (!current)
+            break;
+
+        del = current;
+    }
+    list = nullptr;
+}
+
+void StringListAdd(char** list, String str)
+{
+    if (list[0] == nullptr)
+    {
+        // adding first node
+        list[1] = static_cast<char*>
+                (malloc(sizeof(char**) * 2));
```

Comment:
malloc returning a pointer or null. you need check it

Reply:
It is checked in line 63:
if(!list[1]) 
   ///